### PR TITLE
Add liveness and readiness probes

### PR DESCRIPTION
**Pull request by tolusha**
Signed-off-by: Anatolii Bazko <abazko@redhat.com>

### Reference issue
https://issues.redhat.com/browse/CRW-683
https://issues.redhat.com/browse/CRW-916

### What does this PR do
- add liveness and readiness probes for operator
- add liveness probe for keycloak and postgres containers

### CODE REVIEW

File: cmd/manager/main.go [hunk 1]
```diff
@@ -155,6 +160,16 @@ func main() {
 		os.Exit(1)
 	}
 
+	// Setup health checks
+	if err := mgr.AddHealthzCheck("healthz", healthz.Ping); err != nil {
+		log.Error(err, "Unable to set up health check")
+		os.Exit(1)
+	}
+	if err := mgr.AddReadyzCheck("readyz", healthz.Ping); err != nil {
```
Comment (sleshchenko) [inline cmd/manager/main.go:168]: I understand why Keycloak and Postgres, registries have readiness probe

> Readiness
> Readiness probes are designed to let Kubernetes know when your app is ready to serve traffic. Kubernetes makes sure the readiness probe passes before allowing a service to send traffic to the pod. If a readiness probe starts to fail, Kubernetes stops sending traffic to the pod until it passes.

But as far as I understand Che Operator does not provide any REST API.
So, I'm missing some point or the readiness probe is redundant.

The readiness probe won't affect metrics, right https://github.com/eclipse/che-operator/blob/probes/deploy/olm-catalog/eclipse-che-preview-kubernetes/manifests/che-operator.clusterserviceversion.yaml#L372 ?

Comment (tolusha) [reply cmd/manager/main.go:168]: > But as far as I understand Che Operator does not provide any REST API.

`mgr.AddReadyzCheck("readyz", healthz.Ping)` does the trick.

> The readiness probe won't affect metrics, right

It shouldn't. Do you have any concerns ?

**Comment by tolusha** [comment]
OpenShift tests fail due to fact that it tries to use the `quay.io/eclipse/che-operator:nightly` image where readiness and liveness probes aren't added.

**Comment by tolusha** [comment]
Tested locally on OpenShift, it works fine.

**Review by mmorhun — APPROVED** [review]

**Review by sleshchenko — APPROVED** [review]

**Pull request merged**